src/gui/windows/creative_inventory.zig

```diff
@@ -40,38 +45,91 @@ fn lessThan(_: void, lhs: Item, rhs: Item) bool {
 }
 
 pub fn onOpen() void {
-	items = .init(main.globalAllocator);
-	var itemIterator = main.items.iterator();
-	while(itemIterator.next()) |item| {
-		items.append(Item{.baseItem = item.*});
-	}
-	std.mem.sort(Item, items.items, {}, lessThan);
-	inventory = Inventory.init(main.globalAllocator, items.items.len, .creative, .other);
-	for(0..items.items.len) |i| {
-		inventory.fillAmountFromCreative(@intCast(i), items.items[i], 1);
-	}
+	searchString = "";
+	initContent();
+}
+
+pub fn onClose() void {
+	deinitContent();
+	main.globalAllocator.free(searchString);
+}
 
-	const list = VerticalList.init(.{padding, padding + 16}, 140, 0);
-	var i: u32 = 0;
-	while(i < items.items.len) {
+fn initContent() void {
+	const root = VerticalList.init(.{padding, padding}, 300, 0);
+	{
+		const list = VerticalList.init(.{0, padding + padding}, 48, 0);
 		const row = HorizontalList.init();
-		for(0..10) |_| {
-			if(i >= items.items.len) break;
-			row.add(ItemSlot.init(.{0, 0}, inventory, i, .default, .takeOnly));
-			i += 1;
-		}
+		const label = Label.init(.{0, 3}, 56, "Search:", .right);
+
+		searchInput = TextInput.init(.{0, 0}, 288, 22, searchString, .{.callback = &filter}, .{});
+
+		row.add(label);
+		row.add(searchInput);
 		list.add(row);
+		list.finish(.center);
+		root.add(list);
+	}
+	{
+		const list = VerticalList.init(.{0, padding}, 144, 0);
+		items = .init(main.globalAllocator);
+		var itemIterator = main.items.iterator();
+		while(itemIterator.next()) |item| {
+			if(searchString.len != 0 and std.mem.indexOf(u8, item.id(), searchString) == null) continue;
```

Comment:
Since you are interested in whether it contains the value and not the index, you should use `std.mem.containsAtLeast`